activerecord/test/cases/batches_test.rb

```diff
@@ -435,6 +462,25 @@ def test_in_batches_should_end_at_the_finish_option
     end
   end
 
+  def test_in_batches_should_end_at_the_finish_option_when_implicit_order_column_is_configured
+    old_implicit_order_column = Post.implicit_order_column
+    Post.implicit_order_column = "title"
+
+    sorted_posts = Post.order("title ASC")
+    post = Post.find(6)
+    expected_query_count = sorted_posts.index(post) + 3 # The number of batches, plus first, last, and accounting for the 0 based index.
+
+    # The number of queries is calculated because of the differences in alphanumeric sorting between DB types.
```

Comment:
If order is important (which I think it is here), how about testing each database separately?